libs/shared-atomic-design-components/src/lib/atoms/CardTags.tsx

```diff
@@ -1,65 +1,114 @@
-import { Tag } from 'react-bulma-components'
 import './CardTags.scss'
+import React, { useState, useEffect } from 'react'
+import { debounce } from 'lodash'
+import { useMediaQuery } from '@material-ui/core'
+import { log } from 'console'
 
 export interface CardTagsProps {
   items: string[]
   shortList?: boolean
   formatter?: (item: string) => string
 }
 
-interface CardTagProps {
-  children: string
-  className?: string
-  key: string
-}
+const CardTags = ({ items, shortList = false, formatter }: CardTagsProps) => {
+  const [visibleChips, setVisibleChips] = useState([])
+  const [remainingChips, setRemainingChips] = useState([])
+
+  const parentRefDesktop = React.useRef<HTMLDivElement>()
+  const parentRefMobile = React.useRef<HTMLDivElement>()
+
+  const isMobile = useMediaQuery('(max-width:768px)')
 
-const CardTag = ({ children, className }: CardTagProps) => (
-  <>
-    <Tag
-      className={className}
-      size="normal"
-      textWeight="bold"
-      responsive={{ tablet: { hide: { value: true } } }}
-    >
-      {children}
-    </Tag>
-    <Tag
-      className={className}
-      size="medium"
-      textWeight="bold"
-      responsive={{ mobile: { hide: { value: true } } }}
-    >
-      {children}
-    </Tag>
-  </>
-)
-
-const CardTags = ({ items, shortList, formatter }: CardTagsProps) => {
   const additionalTagsCount = items.length - 3
-  const tagList = shortList ? items.slice(0, 3) : items
-  const hasAdditionalTags = shortList && additionalTagsCount > 0
+  const itemsShowMobile = items.map(formatter).slice(0, 3)
+  const hasAdditionalTags = (shortList || isMobile) && additionalTagsCount > 0
+
+  useEffect(() => {
+    const formattedItems = items.map(formatter)
+
+    const calculateChips = () => {
+      const parentWidth = parentRefDesktop.current?.offsetWidth
+      const gapSize = 16 // Adjust this based on the design (16px gap)
+      const sidePadding = 12 // Adjust this based on the design (12px padding each side )
+      const chipTotalMargin = 2 * sidePadding + gapSize
+
+      const chipWidths = formattedItems.map((chip) => {
+        const chipSpan = document.createElement('span')
+        chipSpan.style.visibility = 'hidden'
+        chipSpan.classList.add('chip')
+        chipSpan.innerHTML = chip
+
+        parentRefDesktop.current.appendChild(chipSpan)
```

Comment:
We might want to use the optional chaining also here, as you're doing on line 30:

```suggestion
        parentRefDesktop.current?.appendChild(chipSpan)
```